aviary/subsystems/propulsion/utils.py

```diff
@@ -104,6 +107,54 @@ def convert_geopotential_altitude(altitude):
     return altitude
 
 
+def build_engine_deck(aviary_options: AviaryValues):
+    '''
+    Creates an EngineDeck using avaliable inputs and options in aviary_options.
+
+    Parameter
+    ----------
+    aviary_options : AviaryValues
+        Options to use in creation of EngineDecks.
+
+    Returns
+    ----------
+    engine_models : <list of EngineDecks>
+        List of EngineDecks created using provided aviary_options.
+    '''
+    # import locally placed to avoid circular import
+    from aviary.subsystems.propulsion.engine_deck import EngineDeck
+
+    # Build a single engine deck, currently ignoring vectorization
+    # of AviaryValues (use first index)
+    # TODO build test to verify this works as expected
+    engine_options = AviaryValues()
+    for entry in Aircraft.Engine.__dict__:
+        var = getattr(Aircraft.Engine, entry)
+        # check if this variable exist with useable metadata
+        try:
+            units = _MetaData[var]['units']
+            try:
+                # add value from aviary_options to engine_options
+                aviary_val = aviary_options.get_val(var, units)
+                if isinstance(aviary_val, np.ndarray):
+                    # "Convert" numpy types to standard Python types. Wrap first
```

Comment:
Are we ever going to have an engine model that has a true numpy vector has one of the input/options? if so, then grabbing the 0th element won't be the right thing to do.

Reply:
...I forgot to wrap the 0th index in a numpy array before calling `item()`, even though the comment says to.
That will work for any arbitrary numpy array of any dtype. I think your comment was targeting something else though